### ISSUE TITLE
Would you please tell me what to do if I want to test my own single image~

### ISSUE THREAD

**Issue by kimwao** [researchmm/CKDN#7]
Thanks for your excellent work!
I would like to ask whether only the training phase needs LR as a reference, and a single test should only need to provide SR as input.
But when I used predict_score.py, I found that it refers to “timm.data.dataset”， It still needs to deal with data like training or val phase.
What should I do if I want to test my own pictures, or what should be the input form?
I'd appreciate it if you would tell me~

**Comment by sea-with-stars** [comment]
Yes, I met the same bug, but I cannot fix it up.

**Comment by sea-with-stars** [comment]
my code maybe work>?`"""Sample PyTorch Inference script for two pairs of images"""

from __future__ import absolute_import
from __future__ import division
from __future__ import print_function

import os
import time
import argparse
import logging
import numpy as np
import torch
from PIL import Image

from timm.data import create_transform
from timm.utils import AverageMeter, setup_default_logging
import ckdn

torch.backends.cudnn.benchmark = True

parser = argparse.ArgumentParser(description='PyTorch Two Pairs of Images Inference')
parser.add_argument('degraded_path', metavar='DEGRADED1',default="data_folder/degraded/test1_00_02.png",nargs='?',
                    help='path to the first degraded image')
parser.add_argument('restored_path', metavar='RESTORED1',default="data_folder/restored/test1_00_02.png",nargs='?',
                    help='path to the first restored image')


parser.add_argument('--output_dir', metavar='DIR', default='./',
                    help='path to output files')
parser.add_argument('--model', '-m', metavar='MODEL', default='resnet',
                    help='model architecture (default: dpn92)')
parser.add_argument('--img-size', default=288, type=int,
                    metavar='N', help='Input image dimension')
parser.add_argument('--interpolation', default='', type=str, metavar='NAME',
                    help='Image resize interpolation type (overrides model)')
parser.add_argument('--num-classes', type=int, default=1000,
                    help='Number classes in dataset')
parser.add_argument('--checkpoint', default='./model_best.pth.tar', type=str, metavar='PATH',
                    help='path to latest checkpoint (default: none)')
parser.add_argument('--pretrained', dest='pretrained', action='store_true',
                    help='use pre-trained model')
parser.add_argument('--num-gpu', type=int, default=1,
                    help='Number of GPUS to use')
parser.add_argument('--no-test-pool', dest='no_test_pool', action='store_true',
                    help='disable test time pool')
parser.add_argument('--topk', default=5, type=int,
                    metavar='N', help='Top-k to output to CSV')


def main():
    setup_default_logging()
    args = parser.parse_args()
    # might as well try to do something useful...
    args.pretrained = args.pretrained or not args.checkpoint

    # create model
    model = ckdn.model()
    logging.info('Model %s created, param count: %d' %
                 (args.model, sum([m.numel() for m in model.parameters()])))

    state_dicts = torch.load(args.checkpoint)['state_dict']
    new_state_dict = {}
    for k in state_dicts.keys():
        new_state_dict[k.split('module.')[1]] = state_dicts[k]

    model.load_state_dict(new_state_dict, strict=False)

    if args.num_gpu > 1:
        model = torch.nn.DataParallel(model, device_ids=list(range(args.num_gpu))).cuda()
    else:
        model = model.cuda()

    model.eval()

    # Load and preprocess the first pair of images
    degraded1 = Image.open(args.degraded_path).convert('RGB')
    restored1 = Image.open(args.restored_path).convert('RGB')
    transform = create_transform(
        input_size=args.img_size,
        interpolation=args.interpolation,
        crop_pct=0.875
    )
    degraded1 = transform(degraded1).unsqueeze(0)
    restored1 = transform(restored1).unsqueeze(0)


    # Concatenate the two pairs of images into a batch
    input1 = torch.cat([degraded1, restored1], dim=1)
    input = torch.cat([input1,input1], dim=1).cuda()

    with torch.no_grad():
        output = model(input)
        outputs = output[:, 0].cpu().numpy()

    with open(os.path.join(args.output_dir, './output.txt'), 'w') as out_file:
        filenames = [os.path.basename(args.degraded_path), os.path.basename(args.degraded_path)]
        for filename, output in zip(filenames, [str(o) for o in outputs]):
            out_file.write('{0},{1}\n'.format(filename, output))


if __name__ == '__main__':
    main()`
